debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnector.java

```diff
@@ -56,4 +67,54 @@ public void stop() {
     public ConfigDef config() {
         return SqlServerConnectorConfig.configDef();
     }
+
+    @Override
+    public Config validate(Map<String, String> connectorConfigs) {
+        Configuration config = Configuration.from(connectorConfigs);
+        SqlServerConnectorConfig sqlServerConfig = new SqlServerConnectorConfig(config);
+        // First, validate all of the individual fields, which is easy since don't make any of the fields invisible ...
+        Map<String, ConfigValue> results = config.validate(SqlServerConnectorConfig.ALL_FIELDS);
+
+        // Get the config values for each of the connection-related fields ...
+        ConfigValue hostnameValue = results.get(SqlServerConnectorConfig.HOSTNAME.name());
+        ConfigValue portValue = results.get(SqlServerConnectorConfig.PORT.name());
+        ConfigValue databaseValue = results.get(SqlServerConnectorConfig.DATABASE_NAME.name());
+        ConfigValue userValue = results.get(SqlServerConnectorConfig.USER.name());
+        ConfigValue passwordValue = results.get(SqlServerConnectorConfig.PASSWORD.name());
+        final String passwordStringValue = config.getString(SqlServerConnectorConfig.PASSWORD);
+
+        if (Strings.isNullOrEmpty(passwordStringValue)) {
+            LOGGER.debug("The connection password is empty");
+        }
+
+        // If there are no errors on any of these ...
+        if (hostnameValue.errorMessages().isEmpty()
+                && portValue.errorMessages().isEmpty()
+                && userValue.errorMessages().isEmpty()
+                && passwordValue.errorMessages().isEmpty()
+                && databaseValue.errorMessages().isEmpty()) {
+            // Try to connect to the database ...
+            try (SqlServerConnection connection = new SqlServerConnection(sqlServerConfig.jdbcConfig(), Clock.system(),
+                    sqlServerConfig.getSourceTimestampMode(), null)) {
+                // SqlServerConnection will try retrieving database, no need to run another query.
+                LOGGER.debug("Successfully tested connection for {} with user '{}'", connection.connectionString(),
+                        connection.username());
+            }
+            catch (Throwable e) {
+                LOGGER.debug("Failed testing connection for {} with user '{}'", sqlServerConfig.jdbcConfig(),
```

Comment:
`Configuration.withMaskedPasswords()` should be used when logging the configuration.